src/Mod/PartDesign/Gui/TaskRevolutionParameters.cpp

```diff
@@ -357,7 +357,10 @@ void TaskRevolutionParameters::updateUI(int index)
 void TaskRevolutionParameters::onSelectionChanged(const Gui::SelectionChanges& msg)
 {
     if (msg.Type == Gui::SelectionChanges::AddSelection) {
-        if (selectionFace) {
+        int mode = ui->changeMode->currentIndex();
+        if (static_cast<PartDesign::Revolution::RevolMethod>(mode)
+                == PartDesign::Revolution::RevolMethod::ToFace
+            && selectionFace) {

```

Comment:
This change seems like it shouldn't be needed. @Syres916 could you share your thought process here?

Reply:
I just tested without this addition and just the `updateUI`. That works initially, but fails to work perfectly well after a few attempts.

Reply:
Well it was seemed to make sense (at least to me) that if selected face was already true then only with To Face mode would it be linked but as I've stated I don't understand what you were actually trying to achieve with the original PR. If it's going to help, I'll close this and just let you bring forward a proper fix.

Reply:
Please don't close.

The original PR was to add support for certain other means of rotation, like up to a certain face. That needed a large overhaul of the entire tool, so regressions and issues got introduced. It is actually surprising that axis selection was affected because that was one of the few things not touched (see "onAxisChanged").

Reply:
OK with this change the issue can still happen if we are in the "up to face" mode and want to change the revolution axis. I think making `selectionFace` false outside the context might be a better option, see #11822.